dask/dataframe/tests/test_shuffle.py

```diff
@@ -179,22 +184,53 @@ def test_partitioning_index_categorical_on_values():
     "npartitions", [1, 4, 7, pytest.param(23, marks=pytest.mark.slow)]
 )
 def test_set_index_general(npartitions, shuffle_method):
+    names = ["alice", "bob", "ricky"]
     df = pd.DataFrame(
-        {"x": np.random.random(100), "y": np.random.random(100) // 0.2},
+        {
+            "x": np.random.random(100),
+            "y": np.random.random(100) // 0.2,
+            "z": np.random.choice(names, 100),
+        },
         index=np.random.random(100),
     )
+    # Ensure extension dtypes work
+    # NOTE: Older version of pandas have known issues with extension dtypes.
```

Comment:
The big issue was that, prior to `pandas=1.4`, index objects couldn't hold arbitrary extension types (they would often fall back to `object`) https://pandas.pydata.org/pandas-docs/stable/whatsnew/v1.4.0.html#index-can-hold-arbitrary-extensionarrays. There were also a handful of small bugs I encountered along the way that have since been resolved in `pandas`.